packages/compiler-dom/src/transforms/vOn.ts

```diff
@@ -24,28 +24,56 @@ const isKeyboardEvent = /*#__PURE__*/ makeMap(
   true
 )
 
+const generateModifiers = (modifiers: string[]) => {
+  const modifiersSize = modifiers.length
+
+  const keyModifiers = []
+  const nonKeyModifiers = []
+  const eventOptionModifiers = []
+
+  for (let i = 0; i < modifiersSize; i++) {
+    const modifier = modifiers[i]
+
+    if (isEventOptionModifier(modifier)) {
+      eventOptionModifiers.push(modifier)
+    } else {
+      if (isNonKeyModifier(modifier)) {
+        nonKeyModifiers.push(modifier)
+      } else {
+        keyModifiers.push(modifier)
+      }
+    }
+  }
+
+  return {
+    runtimeModifiers: [keyModifiers, nonKeyModifiers],
```

Comment:
How about avoiding `runtimeModifiers` altogether and just return `{ keyModifiers, nonKeyModifiers, eventOptionsModifiers }`?

Reply:
return  [keyModifiers, nonKeyModifiers, eventOptionsModifiers]
The const [keyModifiers, nonKeyModifiers, eventOptionsModifiers] = returnedArray
would be better.

Reply:
My fear is losing the context of "runtimeModifiers". Is there a problem with that?